modules/installation-special-config-storage.adoc

```diff
@@ -0,0 +1,155 @@
+
+. If you want to use TPM v2 to encrypt your cluster, check to see if TPM v2 encryption needs to be enabled in the BIOS on each node. This is required on most Dell systems. Check the manual for your computer.
+
+. If you want to use Tang to encrypt your cluster, follow these preparatory steps:
+
+.. Access a Red Hat Enterprise Linux server from which you can configure the encryption settings.
+
+.. Set up or access an existing Tang server. See link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption_security-hardening#network-bound-disk-encryption_configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption[Network-bound disk encryption] for instructions. See link:https://youtu.be/2uLKvB8Z5D0[Securing Automated Decryption New Cryptography and Techniques] for a presentation on Tang.
+
+.. Install the `clevis` package, if it is not already installed:
++
+[source,terminal]
+----
+$ sudo yum install clevis -y
+----
+
+.. Generate a thumbprint for the Tang server.
+
+... In the following command, replace the value of `url` with the Tang server URL:
++
+[source,terminal]
+----
+$ echo nifty random wordwords | clevis-encrypt-tang '{"url":"https://tang.example.org"}'
+----
++
+.Example output
+[source,terminal]
+----
+The advertisement contains the following signing keys:
+
+PLjNyRdGw03zlRoGjQYMahSZGu9
+----
+
+... When the `Do you wish to trust these keys? [ynYN]` prompt displays, type `Y`, and the thumbprint is displayed:
++
+.Example output
+[source,terminal]
+----
+eyJhbmc3SlRyMXpPenc3ajhEQ01tZVJiTi1oM...
+----
+
+.. If nodes are configured with static IP addressing, ensure you use the coreos-installer `--append-karg` option when installing {op-system} nodes to set the IP address of the installed system.  Append `rd.neednet=1`, plus the `ip=` and other arguments needed for your network.
++
+[IMPORTANT]
+====
+Some methods for configuring static IPs do not affect the initramfs after the first boot, and so will not work with Tang encryption.  These include the coreos-installer `--copy-network` option, and adding `ip=` arguments to the kernel command line of the live ISO or PXE image during installation.  Incorrect static IP configuration causes the second boot of the node to fail.
+====
+
+. Change to the directory that contains the installation program and generate the Kubernetes manifests for the cluster:
++
+[source,terminal]
+----
+$ ./openshift-install create manifests --dir=<installation_directory>
+----
+
+. Create a Butane config configuring disk encryption and/or mirroring. For example, to configure storage for worker nodes, create `$HOME/clusterconfig/worker-storage.bu`:
++
+[source,yaml]
+.Butane config example
+----
+variant: openshift
+version: 4.8.0
+metadata:
+  name: worker-storage <1>
+  labels:
+    machineconfiguration.openshift.io/role: worker <1>
+boot_device:
```

Comment:
Should probably include the `layout` key as we've mentioned above that this feature is available on `x86_64 nodes booted with BIOS or UEFI and on ppc64le nodes.`

Reply:
Good catch.  Updated.